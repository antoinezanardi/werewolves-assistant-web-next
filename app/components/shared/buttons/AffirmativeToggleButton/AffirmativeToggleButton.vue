<template>
  <PrimeVueToggleButton
    id="affirmative-toggle-button"
    v-model="model"
    off-icon="fa fa-thumbs-down"
    :off-label="$t('shared.no')"
    on-icon="fa fa-thumbs-up"
    :on-label="$t('shared.yes')"
  />
</template>

<script setup lang="ts">
const model = defineModel<boolean>();
</script>