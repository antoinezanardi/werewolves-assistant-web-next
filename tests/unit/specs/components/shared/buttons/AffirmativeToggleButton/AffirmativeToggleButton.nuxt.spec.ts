import type { mount } from "@vue/test-utils";
import type { ComponentMountingOptions } from "@vue/test-utils/dist/mount";
import type ToggleButton from "primevue/togglebutton";

import AffirmativeToggleButton from "~/components/shared/buttons/AffirmativeToggleButton/AffirmativeToggleButton.vue";
import { mountSuspendedComponent } from "@tests/unit/utils/helpers/mount.helpers";

describe("Affirmative Toggle Button Component", () => {
  let wrapper: ReturnType<typeof mount<typeof AffirmativeToggleButton>>;

  async function mountAffirmativeToggleButtonComponent(options: ComponentMountingOptions<typeof AffirmativeToggleButton> = {}):
  Promise<ReturnType<typeof mount<typeof AffirmativeToggleButton>>> {
    return mountSuspendedComponent(AffirmativeToggleButton, {
      props: { modelValue: true },
      ...options,
    });
  }

  beforeEach(async() => {
    wrapper = await mountAffirmativeToggleButtonComponent();
  });

  it("should match snapshot when rendered.", () => {
    expect(wrapper).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe("Toggle Button", () => {
    it("should pass v-model value to the toggle button when rendered.", () => {
      const toggleButton = wrapper.findComponent<typeof ToggleButton>("#affirmative-toggle-button");

      expect(toggleButton.attributes("modelvalue")).toBe("true");
    });

    it("should change v-model value when toggle button emits input event.", async() => {
      wrapper = await mountAffirmativeToggleButtonComponent({ shallow: false });
      const toggleButtonCheckbox = wrapper.find<HTMLInputElement>("#affirmative-toggle-button");
      await toggleButtonCheckbox.trigger("click");

      expect(toggleButtonCheckbox.attributes("value")).toBeUndefined();
    });

    it("should translate no label when rendered.", () => {
      const toggleButton = wrapper.findComponent<typeof ToggleButton>("#affirmative-toggle-button");

      expect(toggleButton.attributes("off-label")).toBe("No");
    });
  });
});