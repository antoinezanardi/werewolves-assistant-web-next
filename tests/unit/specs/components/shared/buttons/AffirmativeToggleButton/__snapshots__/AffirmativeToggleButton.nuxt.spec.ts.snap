// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Affirmative Toggle Button Component > should match snapshot when rendered. 1`] = `"<toggle-button-stub id="affirmative-toggle-button" modelvalue="true" off-icon="fa fa-thumbs-down !text-white" off-label="No"></toggle-button-stub>"`;
